nshlib/nsh_envcmds.c

```diff
@@ -552,7 +552,7 @@ int cmd_unset(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
   /* Unset environment variable */
 
   status = unsetenv(argv[1]);
-  if (status < 0)
+  if (status < 0 && errno != ENOENT)
```

Comment:
Do we need to try `unsetenv` again in case of `ENOENT`? Because now if `unsetenv` returns with `ENOENT` the `cmd_unset` returns `OK`